composer/trainer/trainer_hparams.py

```diff
@@ -646,3 +770,367 @@ def load(cls, model: str) -> TrainerHparams:
+        default=None,
+    )
+
+    def initialize_object(self, model: ComposerModel, dataloader_hparams: DataLoaderHparams) -> EvalKwargs:
+        """Construct a kwargs dictionary that can be unpacked and passed into :meth:`.Trainer.eval`.
+
+        Args:
+            model (ComposerModel): The model.
+            dataloader_hparams (DataLoaderHparams): The dataloader hyperparameters.
+
+        Returns:
+            EvalKwargs: A kwargs dictionary that can be unpacked and passed into :meth:`.Trainer.eval`.
+        """
+        # Dataloader
+        dataloader = _initialize_dataloader(
+            dataset_hparams=self.dataset,
+            dataloader_label=self.dataloader_label,
+            batch_size=self.batch_size,
+            subset_num_batches=self.subset_num_batches,
+            dataloader_hparams=dataloader_hparams,
+        )
+        assert dataloader is not None, "The dataloader is a required argument"
+
+        # Metrics
+
+        # TODO(Ravi): Cleanup this code as part of the MetricsModule. This code was copied
+        # from composer/datasets/evaluator.py, but will likely be removed when the MetricsModule
+        # is implemented, as the trainer will not be responsible for constructing metrics.
+
+        # Get and copy all the model's associated evaluation metrics
+        model_metrics = model.metrics(train=False)
+        if isinstance(model_metrics, Metric):
+            # Forcing metrics to be a MetricCollection simplifies logging results
+            model_metrics = MetricCollection([model_metrics])
+
+        # Use all the metrics from the model if no metric_names are specified
+        if self.metric_names is None:
+            metrics = copy.deepcopy(model_metrics)
+        else:
+            metrics = MetricCollection([])
+            for metric_name in self.metric_names:
+                try:
+                    metric = model_metrics[metric_name]
+                except KeyError as e:
+                    raise RuntimeError((f"No metric found with the name {metric_name}. Check if this "
+                                        "metric is compatible/listed in your model metrics. ")) from e
+                assert isinstance(metric, Metric), "all values of a MetricCollection.__getitem__ should be a metric"
+                metrics.add_metrics(copy.deepcopy(metric))
+            if len(metrics) == 0:
+                raise RuntimeError(("No metrics compatible with your model were added to this evaluator. "
+                                    "Check that the metrics you specified are compatible/listed in your model."))
+
+        return {
+            "dataloader": dataloader,
+            "dataloader_label": self.dataloader_label,
+            "metrics": metrics,
+            "subset_num_batches": self.subset_num_batches,
+            "log_level": self.log_level,
+        }
+
+
+@dataclasses.dataclass
+class ExperimentHparams(hp.Hparams):
```

Comment:
I originally thought about doing it this way, but was confused on how to break it down. I was torn on where `FitKwargs`, `EvalKwargs`, `FitHparams`, and `EvalHparams` live. Since they are directly tied to the trainer, I was thinking that `TrainerHparams` made more sense, but they are only used by the `ExperimentHparams` :shrug:

Reply:
(Resolved offline)